### PR TITLE
Add some of my favourite papers on unikernels

### PR DESCRIPTION
**Pull request by nderjung**
Hi! 👋

I'm a PhD student at Lancaster University and I research and work on unikernels.  After finding `papers-we-love` and seeing `unikernels` subdirectory, I was quick to fork and add some of my favourite papers that I've read on unikernels.  In this commit, I introduce 53 links to pdfs in the public domain or, at the very least, links to the publisher's public citation page of the peer reviewed paper.  

The papers have been organised in chronology following the influence of unikernels, particularly focusing on papers which identify traits in increasing application security, portability and performance with respect to its runtime.  This follows from the early definitions, such as microkernels which attempt to increase application portability by only facilitating reliable Inter-Process Communication (IPC).  Second-wave follows the centralizastion of commodity hardware into 'the cloud' and the realization of increasing application performance with networked, mult-core, many user systems.

In the unikernels section, I start with links to a number of papers which describe the concept of unikernels and the need for increasing performance in the cloud and how well tuned Virtual Machines (VMs) can aid in increasing performance.  Following this section is a link to relevant published articles of a number of unikernel/libOS implementations. There are a number of links to articles which compare the performance of unikernels themselves to similar counterparts including monolithic OSes and containers. Finally, unikernels have been used in a number of interesting scenarios, and the applications of unikernels section has links to papers which implement the unikernel model in various scenarios following the claims of their performance and security benefits.

### CODE REVIEW

**Comment by NewAlexandria** [comment]
great write-up

File: unikernels/README.md [hunk 1]
```diff
@@ -1,9 +1,78 @@
-## Unikernels (LibOS)
+# Unikernels
+
+## First-wave: lightweight kernels (microkernels)
+
+* [HYDRA: the kernel of a multiprocessor operating system](https://kilthub.cmu.edu/articles/HYDRA_--_the_kernel_of_a_multiprocessor_operating_system/6606224/files/12096734.pdf)
+* [Two years of experience with a μ-Kernel based OS](http://delivery.acm.org/10.1145/130000/122124/p51-liedtke.pdf)
```
Comment (NewAlexandria) [inline unikernels/README.md]: This link is broken

Comment (NewAlexandria) [inline unikernels/README.md]: this link is broken

File: unikernels/README.md [hunk 2]
```diff
@@ -1,9 +1,78 @@
...
+* [Arrakis: The Operating System Is the Control Plane](https://dl.acm.org/citation.cfm?id=2812806)
+* [Embracing diversity in the Barrelfish manycore operating system](http://www.barrelfish.org/publications/barrelfish_mmcs08.pdf)
+* [Decoupling Cores, Kernels, and Operating Systems](https://www.usenix.org/system/files/conference/osdi14/osdi14-paper-zellweger.pdf)
+
+## Unikernels: bespoke, compile-time specialization based on a libOS
+
+### Introduction
 
 * [Unikernels: Library Operating Systems for the Cloud](http://anil.recoil.org/papers/2013-asplos-mirage.pdf)
+* [Unikernels: the rise of the virtual library operating system](http://delivery.acm.org/10.1145/2570000/2566628/p30-madhavapeddy.pdf)
```
Comment (NewAlexandria) [inline unikernels/README.md]: link is broken

File: unikernels/README.md [hunk 3]
```diff
@@ -1,9 +1,78 @@
...
+### Specific libOS implementations
+
 * [IncludeOS: A minimal, resource efficient unikernel for cloud services](http://folk.uio.no/paalee/publications/2015-cloudcom.pdf)
+* [OSv—optimizing the operating system for virtual machines](https://www.usenix.org/system/files/conference/atc14/atc14-paper-kivity.pdf)
+* [ClickOS and the art of network function virtualization](https://www.usenix.org/system/files/conference/nsdi14/nsdi14-paper-martins.pdf)
+* [Enabling fast, dynamic network processing with clickos](http://conferences.sigcomm.org/sigcomm/2013/papers/hotsdn/p67.pdf)
 * [The rump kernel: A tool for driver development and a toolkit for applications](https://www.netbsd.org/gallery/presentations/justin/2015_AsiaBSDCon/justincormack-abc2015.pdf)
+* [Flexible operating system internals: the design and implementation of the anykernel and rump kernels](https://aaltodoc.aalto.fi/bitstream/handle/123456789/6318/isbn9789526049175.pdf?sequence=1)
+* [Rump Kernels: No OS? No Problem!](https://www.usenix.org/system/files/login/articles/login_1410_03_kantee.pdf)
+* [HermitCore: A Unikernel for Extreme Scale Computing](http://delivery.acm.org/10.1145/2940000/2931093/a4-lankes.pdf)
```
Comment (NewAlexandria) [inline unikernels/README.md]: link is broken

**Comment by NewAlexandria** [comment]
The repo of links is a really great collection IMO.  IDK if anyone is against adding 'so many' new links at once.   Without anyone to better vett if these are all 'ideal' papers to add, I think this is a good addition to PWL.

**Comment by DarrenN** [comment]
@nderjung thanks for putting this PR together! I'll merge this as soon as the broken links are corrected. Thanks again.

**Comment by nderjung** [comment]
My mistake on some of the links, I have access to the pdfs and didn't catch all of the ones that weren't publicly accessible!  Should be fixed now :)

**Pull request merged**